jib-core/src/main/java/com/google/cloud/tools/jib/builder/BuildContainerConfigurationStep.java

```diff
@@ -112,17 +102,8 @@ private BlobDescriptor afterBaseImageLayerFuturesFuture()
       CountingDigestOutputStream digestOutputStream =
           new CountingDigestOutputStream(ByteStreams.nullOutputStream());
       containerConfigurationBlob.writeTo(digestOutputStream);
-      BlobDescriptor containerConfigurationBlobDescriptor = digestOutputStream.toBlobDescriptor();
-
-      timer.lap(
-          "Pushing container configuration " + containerConfigurationBlobDescriptor.getDigest());
-
-      // TODO: Use PushBlobStep.
-      // Pushes the container configuration.
-      registryClient.pushBlob(
-          containerConfigurationBlobDescriptor.getDigest(), containerConfigurationBlob);
 
-      return containerConfigurationBlobDescriptor;
+      return new BlobAndDigest(containerConfigurationBlob, digestOutputStream.toBlobDescriptor());
```

Comment:
This could probably avoid creating a new tuple class `BlobAndDigest` by just returning the `Blob` here and generating the `Digest` in the push step.

Reply:
Sounds good, I wasn't sure if the Digest generation should be kept in the build step.